### ISSUE TITLE
ms-appx:// references in XAML are not redirected when PublishSingleFile is enabed

### ISSUE THREAD

**Issue by lhak** [microsoft/microsoft-ui-xaml#9468]
### Describe the bug

When PublishSingleFile is enabled, all assets are extracted to a temporary folder. However, all references in XAML to files (e.g. images) using ms-appx:// still point to the application folder and thus do not work.

### Steps to reproduce the bug

1. Create an application with single file deployment
2. Add an image file to the application and reference it with ms-appx://
3. Observe that the file will be used correctly when the app is started from visual studio but not when it is published as a single file

### Expected behavior

All ms-appx:// references should be redirected

### NuGet package version

WinUI 3 - Windows App SDK 1.5.1: 1.5.240311000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by ghost1372** [comment]
You can exclude your assets from publish single file so this can be fixed.
```
<ItemGroup> <Content Update="Plugin.dll"> <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory> <ExcludeFromSingleFile>true</ExcludeFromSingleFile> </Content> </ItemGroup>
```
https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli

But in general, this bug must be fixed

**Comment by Scottj1s** [comment]
@lhak Thanks for reporting this.  

The issue is that the files under your Assets folder are not being copied to the publish output.  This is due to a difference in behavior between copying package output and copying publish output.  For package output, all Assets are copied regardless of each Content element's CopyToOutputDirectory metadata.  However, for publish output, unless CopyToOutputDirectory is specified, the item is not copied.  

There are two ways to resolve this:
1. Ensure your assets all have CopyToOutputDirectory set correctly (e.g., to PreserveNewest)
2. Add a target that automatically applies this setting, as the [WinUI Gallery solution has done](https://github.com/microsoft/WinUI-Gallery/blob/main/WinUIGallery/WinUIGallery.csproj#L296-L314).

**Comment by Scottj1s** [comment]
The target mentioned above (solution #2) is now included in Windows App SDK 1.6.  Closing as resolved.
